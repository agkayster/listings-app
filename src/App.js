// import './App.css';
import Lisitngs from './components/AllLinstings';

function App() {
	return (
		<div>
			<Lisitngs />
		</div>
	);
}

export default App;
