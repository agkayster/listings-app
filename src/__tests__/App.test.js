import { render, screen } from '@testing-library/react';
import App from '../App';
import listings from '../db';

test('renders learn react link', () => {
	render(<App />);
	const linkElement = screen.getByText(/Listings/i);
	expect(linkElement).toBeInTheDocument();
});

test('the listings data is correct', () => {
	expect(listings).toMatchSnapshot();
});

test('the listings data length is correct', () => {
	expect(listings['listings']).toHaveLength(4);
});

for (let i = 0; i < listings['listings'].length; i += 1) {
	test(`listings[${i}] should have properties(id, image, no_of_bedrooms, address, postcode, desc, asking_price, status)`, () => {
		expect(listings['listings'][i]).toHaveProperty('id');
		expect(listings['listings'][i]).toHaveProperty('image');
		expect(listings['listings'][i]).toHaveProperty('no_of_bedrooms');
		expect(listings['listings'][i]).toHaveProperty('address');
		expect(listings['listings'][i]).toHaveProperty('postcode');
		expect(listings['listings'][i]).toHaveProperty('desc');
		expect(listings['listings'][i]).toHaveProperty('asking_price');
		expect(listings['listings'][i]).toHaveProperty('status');
	});
}

test('listings data has postcode 99501 and matches an object', () => {
	const listingA = {
		id: 1,
		image: 'https://media.istockphoto.com/photos/modern-bedroom-interior-stock-photo-picture-id1299098384?b=1&k=20&m=1299098384&s=170667a&w=0&h=Bxi7LTyH5PtX_cS0TEtf7fCUqjQg3VsZSGw3fMlMdio=',
		no_of_bedrooms: 4,
		address: '4262 Geneva Street, Alaska',
		postcode: 99501,
		desc: 'Four bedroom apartment, suitable for a family of 5',
		asking_price: 50,
		status: true,
	};

	expect(listings['listings'][0]).toMatchObject(listingA);
});

test('status to show true twice and false twice', () => {
	expect(listings['listings'].map((listing) => listing.status)).toEqual([
		true,
		false,
		false,
		true,
	]);
});

test('each listing is different', () => {
	expect(listings['listings'][0]).not.toBe(listings['listings'][1]);
	expect(listings['listings'][0]).not.toBe(listings['listings'][2]);
	expect(listings['listings'][0]).not.toBe(listings['listings'][3]);
	expect(listings['listings'][1]).not.toBe(listings['listings'][2]);
	expect(listings['listings'][1]).not.toBe(listings['listings'][3]);
	expect(listings['listings'][2]).not.toBe(listings['listings'][3]);
});
