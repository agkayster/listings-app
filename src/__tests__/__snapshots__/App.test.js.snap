// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the listings data is correct 1`] = `
Object {
  "listings": Array [
    Object {
      "address": "4262 Geneva Street, Alaska",
      "asking_price": 50,
      "desc": "Four bedroom apartment, suitable for a family of 5",
      "id": 1,
      "image": "https://media.istockphoto.com/photos/modern-bedroom-interior-stock-photo-picture-id1299098384?b=1&k=20&m=1299098384&s=170667a&w=0&h=Bxi7LTyH5PtX_cS0TEtf7fCUqjQg3VsZSGw3fMlMdio=",
      "no_of_bedrooms": 4,
      "postcode": 99501,
      "status": true,
    },
    Object {
      "address": "2965 Crowfield Road, Arizona",
      "asking_price": 20,
      "desc": "Two bedroom apartment, facing the skyline. Beautiful view",
      "id": 2,
      "image": "https://images.unsplash.com/photo-1591088398332-8a7791972843?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1374&q=80",
      "no_of_bedrooms": 2,
      "postcode": 85001,
      "status": false,
    },
    Object {
      "address": "851 Cheshire Road, Connecticut",
      "asking_price": 80,
      "desc": "Eight bedroom apartment in the heart of Connecticut. Close to the airport, seaport and schools",
      "id": 3,
      "image": "https://images.unsplash.com/photo-1598928636135-d146006ff4be?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1740&q=80",
      "no_of_bedrooms": 8,
      "postcode": 64005,
      "status": false,
    },
    Object {
      "address": "4647 West Drive, Illinois",
      "asking_price": 120,
      "desc": "Six bedroom apartment in the highbrow area of Chicago, Illinoise. Fit for a large family, lovely scenery and quiet surroundings",
      "id": 4,
      "image": "https://images.unsplash.com/photo-1595268396332-9af381e5935e?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=687&q=80",
      "no_of_bedrooms": 6,
      "postcode": 60631,
      "status": true,
    },
  ],
}
`;
